R/plotFeatureLoadings.R

```diff
@@ -0,0 +1,404 @@
+
+
+#' @rdname plotFeatureLoadings
+#' @export 
+setMethod("plotFeatureLoadings", signature = c(object = "TreeSummarizedExperiment"),
+          function(object,
+                   method = "PCA", plot = "Heatmap",
+                   ...){
+            
+            .check_parameters(object,
+                              method = method, plot = plot, ...)   
+            
+            if (method == "PCA") {
+              loadings_matrix <- attr(reducedDim(tse, "PCA"), "rotation")
+              df <- data.frame(loadings_matrix)
+              df[["Feature"]] <- rownames(loadings_matrix)
+              df[["Feature"]] <- factor(df[["Feature"]], levels = df[["Feature"]])
+              df <- df[ order(abs(df[["PC1"]]), decreasing = TRUE)[1:10], ]
+              df <- df[ order(df[["PC1"]]), ]
+              df <- df[ order(abs(df[["PC2"]]), decreasing = TRUE)[1:10], ]
+              df <- df[ order(df[["PC2"]]), ]
+            } else {
+              stop("These methods are yet to be implemented", call. = FALSE)
+            }
+            
+            if (plot == "Tree") {
+              
+              phylo <- rowTree(object)
+              circ <- ggtree(phylo, layout = "circular")
+              df <- rowData(object)
+              color <- randomcoloR::distinctColorPalette(
+                length(
+                  unique(
+                    df$Phylum
+                  )
+                )
+              )
+              df <- data.frame(Class = df$Phylum)
+              rownames(df) <- phylo$tip.label
+              
+              
+              df2 <- data.frame(abs(loadings_matrix))
+              rownames(df2) <- phylo$tip.label
+              
+              
+              p <- gheatmap(
+                p = circ,
+                data = df,
+                offset = -.1,
+                width = .1,
+                colnames_angle = 95,
+                colnames_offset_y = .5,
+                font.size = 4,
+                color = "black") +
+                ggplot2::scale_fill_manual(
+                  values = color,
+                  name = "Class"
+                )
+              
+              for(i in 1:2){
+                if(i == 1){
+                  p <- p +
+                    ggnewscale::new_scale_fill()
+                }
+                df3 <- dplyr::select(
+                  df2, (all_of(i))
+                )
+                
+                p <- gheatmap(
+                  p,
+                  df3,
+                  offset = i*.065,
+                  width = .1,
+                  colnames_angle = 90,
+                  font.size = 4,
+                  high = "darkslateblue",
+                  low = "gray98",
+                  color = "black",
+                  legend_title = expression(beta[k]))
+              }
+              
+              p
+              
+            } else {
+              .plot_pca_feature_loadings(df, plot)
+            }
```

Comment:
It is hard to see when the function ends. I suggest to save the plot to variable and always end the function with return()